### PR TITLE
scoap3-next: country removal

### PR DESCRIPTION
**Pull request by ErnestaP**
ref: https://github.com/cern-sis/issues-scoap3/issues/173

### CODE REVIEW

File: scripts/country_removal.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                pattern_for_cern_cooperation_agreement = re.compile(r'cooperation agreement with cern', re.IGNORECASE)
+                match_pattern = pattern_for_cern_cooperation_agreement.search(affiliation["value"])
+                if match_pattern is not None:
+                    affiliation.pop("country")
+        except KeyError:
+            print("Author has no affiliations", record['control_number'])
+    return record
+
+def update_records(data):
+    recids = data.keys()
```
Comment (drjova) [inline scripts/country_removal.py]: we don't need the `dois` as it seems, why not just keep an array of `control_numbers` instead?

File: scripts/country_removal.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    updated_records = {}
+    for recid in recids:
+        pid = PersistentIdentifier.get("recid", recid)
+        existing_record = Record.get_record(pid.object_uuid)
+        existing_record = remove_country(existing_record)
+        existing_record.update(dict(existing_record))
+        updated_records[recid] = existing_record
```
Comment (drjova) [inline scripts/country_removal.py]: we don't need to return back the whole records :)

File: scripts/country_removal.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    return records
+
+def remove_country(record):
+    for author in record["authors"]:
+        try:
+            for affiliation in author["affiliations"]:
+                pattern_for_cern_cooperation_agreement = re.compile(r'cooperation agreement with cern', re.IGNORECASE)
+                match_pattern = pattern_for_cern_cooperation_agreement.search(affiliation["value"])
+                if match_pattern is not None:
+                    affiliation.pop("country")
```
Comment (drjova) [inline scripts/country_removal.py]:
```suggestion
                if match_pattern is not None:
                    affiliation.pop("country")
```

I would suggest adding the try/except here so it doesn't fail for the rest of affiliations. Alternatively we could just do:

```python
affiliation.pop('country', None)
```

which doesn't need the `try/catch`

**Comment by drjova** [comment]
@ErnestaP  Let's try it in QA, and please check first how many of these cases we have to compare before and after the script.

**Comment by ErnestaP** [comment]
Done

**Comment by drjova** [comment]
@ErnestaP thank you, we can run it on PROD tomorrow.

**Pull request closed without merging**